### PR TITLE
Implement spectrum copy(), *, *= methods, `--scale` command-line option

### LINKED ISSUE #219: Arbitrary scale factors
For ease of comparison with other codes and normalisation conventions, it would be useful if most CLI outputs had an option to apply an arbitrary scaling factor.

An interesting variation would be to also allow unit conversions. How might those options work together?

### PR DESCRIPTION
**Pull request by ajjackson**
Closes #219 
In order to make the `--scale` option beautiful, tweak the Python API so that any Spectrum can by multiplied by a scalar Real using `*` or `*=`. (These are implemented with "dunder" methods `__mul__` and `__imul__` respectively.)

In practice we will usually use `__imul__`, so makes sense to define `__mul__` using it, creating a fresh Spectrum to write into. In principle it would be more efficient to create one with an "empty" data array, but as there is lots of code making modified array copies, life is a lot simpler with a consistent `copy()` method used as a base for this stuff. If copying the data becomes a noticeable bottleneck we could add an `empty=False` argument that allows us to construct with e.g. `y_data=np.empty_like(self.y_data)*ureg(self.y_data_unit)`, but for now assume YAGNI!

With the new Python/numpy/Pint versions we can use np.copy() directly on an array Quantity, which reduces unit-shuffling boilerplate. :smile: 

To-do
- [x] Refactor *spectra.py* to make full use of `Spectrum.copy()`
- [x] Unit-test `Spectrum.copy()`
- [x] Unit-test `*`, `*=`
- [x] Implement `--scale` argument
- [x] Script-test `--scale` argument
- [x] Update CHANGELOG

### CODE REVIEW

**Comment by ajjackson** [comment]
API wrangling paid off, 11ab6cc was a very satisfying little commit to write :sweat_smile:

**Review by oerc0122 — APPROVED** [review]

File: euphonic/spectra.py [hunk 1]
```diff
@@ -1161,9 +1188,10 @@ def sum(self) -> Spectrum1D:
         metadata.update(self._combine_line_metadata())
         summed_y_data = np.sum(self._y_data, axis=0)*ureg(
             self._internal_y_data_unit).to(self.y_data_unit)
-        return Spectrum1D(self.x_data, summed_y_data,
-                          x_tick_labels=self.x_tick_labels,
-                          metadata=metadata)
+        return Spectrum1D(np.copy(self.x_data),
```
[the remark this answers is not in the document]
Comment (ajjackson) [reply euphonic/spectra.py:1191]: A copy of Spectrum1DCollection would be a Spectrum1DCollection, but this method returns a Spectrum1D.

In principle we could make a copy from e.g. the first element of the collection, but then the constructor is getting called once to make the element from the y_data and again to copy it.

It feels "different enough" to justify breaking the pattern, but I don't have a strong opinion on whether it is _better_ to do it one way or the other.

File: tests_and_analysis/test/euphonic_test/test_spectrum1d.py [hunk 2]
```diff
@@ -502,3 +502,41 @@ def test_add_metadata(self, spectrum_files, metadata,
...
+
+        npt.assert_allclose(spec.y_data * 2.,
+                            (spec * 2.).y_data)
+
+        check_spectrum1d(spec, (spec * 2.) * 0.5)
+
+        with pytest.raises(AssertionError):
+            check_spectrum1d(spec, spec * 2.)
+
+    def test_copy(self):
```
[the remark this answers is not in the document]
Comment (ajjackson) [reply tests_and_analysis/test/euphonic_test/test_spectrum1d.py:517]: Did you have a particular kind of fixture in mind?

I don't see an elegant way to parametrize all of these different operations, but pytest has a lot of features...

[the remark this answers is not in the document]
Comment (ajjackson) [reply tests_and_analysis/test/euphonic_test/test_spectrum1d.py:517]: Could maybe define the check as a kind of teardown operation and make lots of little tests, but that's pesky in its own way.

I could tidy this a bit by defining an `assert_different` function that hides the `pytest.raises`?

[the remark this answers is not in the document]
Comment (ajjackson) [reply tests_and_analysis/test/euphonic_test/test_spectrum1d.py:517]: e.g.
```
    def test_copy(self):
        def assert_different(spectrum, other) -> None:
            with pytest.raises(AssertionError):
                check_spectrum1dcollection(spectrum, other)
            
        spec = get_spectrum1dcollection('gan_bands.json')
        spec.metadata = {'Test': 'item', 'int': 1}

        spec_copy = spec.copy()
        # Copy should be same
        check_spectrum1dcollection(spec, spec_copy)

        # Until data is edited
        spec_copy._y_data *= 2
        assert_different(spec, spec_copy)

        spec_copy = spec.copy()
        spec_copy._x_data *= 2
        assert_different(spec, spec_copy)

        spec_copy = spec.copy()
        spec_copy.x_tick_labels = [(1, 'different')]
        assert_different(spec, spec_copy)

        spec_copy = spec.copy()
        spec_copy.metadata['Test'] = spec_copy.metadata['Test'].upper()
        assert_different(spec, spec_copy)
```

**Comment by ajjackson** [comment]
Rebasing onto the updated tests, not expecting drama but would be better to find out here.

**Review by oerc0122 — APPROVED** [review]

**Pull request merged**